editoast/src/generated_data/error/mod.rs

```diff
@@ -146,86 +148,124 @@ fn generate_errors<Ctx: Default>(
+                &speed_sections::GLOBAL_GENERATORS,
+            );
+            let mut infra_errors = infra_errors.lock().unwrap();
+            infra_errors.extend(result);
+        }),
+        Box::new(|| {
+            let result = generate_errors(
+                ObjectType::Route,
+                infra_cache,
+                &graph,
+                &routes::OBJECT_GENERATORS,
+                &routes::GLOBAL_GENERATORS,
+            );
+            let mut infra_errors = infra_errors.lock().unwrap();
+            infra_errors.extend(result);
+        }),
+        Box::new(|| {
+            let result = generate_errors(
+                ObjectType::SwitchType,
+                infra_cache,
+                &graph,
+                &switch_types::OBJECT_GENERATORS,
+                &[],
+            );
+            let mut infra_errors = infra_errors.lock().unwrap();
+            infra_errors.extend(result);
+        }),
+        Box::new(|| {
+            let result = generate_errors(
+                ObjectType::Detector,
+                infra_cache,
+                &graph,
+                &detectors::OBJECT_GENERATORS,
+                &[],
+            );
+            let mut infra_errors = infra_errors.lock().unwrap();
+            infra_errors.extend(result);
+        }),
+        Box::new(|| {
+            let result = generate_errors(
+                ObjectType::BufferStop,
+                infra_cache,
+                &graph,
+                &buffer_stops::OBJECT_GENERATORS,
+                &buffer_stops::GLOBAL_GENERATORS,
+            );
+            let mut infra_errors = infra_errors.lock().unwrap();
+            infra_errors.extend(result);
+        }),
+        Box::new(|| {
+            let result = generate_errors(
+                ObjectType::OperationalPoint,
+                infra_cache,
+                &graph,
+                &operational_points::OBJECT_GENERATORS,
+                &[],
+            );
+            let mut infra_errors = infra_errors.lock().unwrap();
+            infra_errors.extend(result);
+        }),
+        Box::new(|| {
+            let result = generate_errors(
+                ObjectType::Switch,
+                infra_cache,
+                &graph,
+                &switches::OBJECT_GENERATORS,
+                &[],
+            );
+            let mut infra_errors = infra_errors.lock().unwrap();
+            infra_errors.extend(result);
+        }),
+        Box::new(|| {
+            let result = generate_errors(
+                ObjectType::Catenary,
+                infra_cache,
+                &graph,
+                &catenaries::OBJECT_GENERATORS,
+                &catenaries::GLOBAL_GENERATORS,
+            );
+            let mut infra_errors = infra_errors.lock().unwrap();
+            infra_errors.extend(result);
+        }),
+    ];
+
+    closures.into_par_iter().for_each(|f| f());
+    let infra_errors = infra_errors.lock().unwrap().clone();
```

Comment:
Instead of cloning, you should be able to access directly what's inside the `Mutex`. And see the other comment on `.unwrap()` that will apply similarly here.

```suggestion
    let infra_errors = infra_errors.into_inner().unwrap();
```